code/game/dna/mutations/powers.dm

```diff
@@ -1179,3 +1157,58 @@
 	else
 		H.remoteview_target = null
 		H.reset_perspective()
+
+/datum/mutation/meson_vision
+	name = "Meson Vision"
+	activation_messages = list("More information seems to reach your eyes...")
+	deactivation_messages = list("The amount of information reaching your eyes fades...")
+	instability = GENE_INSTABILITY_MINOR
+	traits_to_add = list(TRAIT_MESON_VISION)
+
+/datum/mutation/meson_vision/New()
+	..()
+	block = GLOB.mesonblock
+
+/datum/mutation/meson_vision/activate(mob/living/M)
+	..()
+	M.update_sight()
+
+/datum/mutation/meson_vision/deactivate(mob/living/M)
+	..()
+	M.update_sight()
+
+/datum/mutation/night_vision
+	name = "Night Vision"
+	activation_messages = list("Were the lights always that bright?")
+	deactivation_messages = list("The ambient light level returns to normal...")
+	instability = GENE_INSTABILITY_MODERATE
+	traits_to_add = list(TRAIT_NIGHT_VISION)
+
+/datum/mutation/night_vision/New()
+	..()
+	block = GLOB.nightvisionblock
+
+/datum/mutation/night_vision/activate(mob/living/M)
+	..()
+	M.update_sight()
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		H.update_misc_effects()
```

Comment:
Okay, I've been looking into this a bit more and I think I might've been wrong about `update_misc_effects()` working for this, to some extent anyway.
What I didn't notice previously is that the `shine()` proc, which is used to check if the eyes should glow or not, checks the `see_in_dark` value of the *eyes organ* rather than the mob.
https://github.com/ParadiseSS13/Paradise/blob/6b326123ac46f4153e1a1ec7ab249e8ea7d64c2a/code/modules/surgery/organs/eyes.dm#L40-L42
The night vision trait only increases the `see_in_dark` value of the *mob* in `update_sight()`, so when it goes to check `shine()` for the overlay, the value on the eyes is unchanged and it always fails.
https://github.com/ParadiseSS13/Paradise/blob/1942cee972ec544d8e47b36a9df609308ec923b6/code/modules/mob/living/carbon/human/species/_species.dm#L908-L910

This would probably best be solved by just editing the variable on their eyes instead, but that could maybe cause issues down the line if it's not reversed properly on `deactivate()`.
I won't be able to test this locally until tomorrow, so I can't offer any other real suggestions on how to fix it at the moment.